src/index.js

```diff
@@ -1,58 +1,74 @@
-import React from "react";
-import PropTypes from "prop-types";
-import { useSelector } from "react-redux";
-import { withRouter } from "react-router-dom";
+import React, { useContext } from 'react';
+import PropTypes from 'prop-types';
 
 /**
- * This function instantiates a React.Component with a set of additional passed props. It supports
- * `Overridable` components.
- *
- * @param {React.Component} Component - the actual component that will be parametrized
- * @param {Object|Function} extraProps - additional properties passed to the final component
+ * Object responsible of keeping track of all overridden components
+ * @constructor object containing the initial map `id: Component` of overriden components
+ */
+class OverriddenComponentsStore {
+  constructor(overriddenCmps) {
+    this.components = overriddenCmps || {};
+  }
+
+  add = (id, Component) => {
+    const alreadyOverridden = id in this.components;
+    if (alreadyOverridden) {
+      throw Error(`The component ${id} has been already overridden.`);
+    }
+    this.components[id] = Component;
+  };
+
+  get = (id) => {
+    return this.components[id];
+  };
```

Comment:
What about making the context dict-like so you can use `overrideStore[id]` to access it? This would allow using a normal dict when using a custom value for the context - especially since the store class is not exported.

If you don't intend to support this, then I'd get rid of the context and use the store directly.

Reply:
See answer below.